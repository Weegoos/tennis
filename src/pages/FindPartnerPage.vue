<template>
  <div>
    <div class="card">
      <p class="text-h4 text-bold">Find a partner</p>
      <q-btn color="primary" label="Place a request" @click="sendRequest" />
    </div>
    <section>
      <q-card class="card">
        <q-card-section>
          <div class="text-h6">Our Changing Planet</div>
          <div class="text-subtitle2">by John Doe</div>
        </q-card-section>
        <q-card-section>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit
        </q-card-section>
      </q-card>
    </section>
    <PostRequest />
  </div>
</template>

<script setup>
import PostRequest from "src/components/Partner/PostRequest.vue";
</script>

<style scoped>
.card {
  margin-right: 200px;
  margin-left: 200px;
  margin-top: 50px;
  margin-bottom: 50px;
}
</style>
