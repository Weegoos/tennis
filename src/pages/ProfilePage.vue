<template>
  <div>Профиль</div>
</template>

<script>
export default {};
</script>

<style></style>
