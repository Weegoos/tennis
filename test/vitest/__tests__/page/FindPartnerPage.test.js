import { installQuasarPlugin } from "@quasar/quasar-app-extension-testing-unit-vitest";
import { mount, shallowMount } from "@vue/test-utils";
import { Quasar } from "quasar";
import FindPartnerPage from "src/pages/FindPartnerPage.vue";
import ProfilePage from "src/pages/ProfilePage.vue";
import { describe, expect, it } from "vitest";

installQuasarPlugin();

describe("tests for FindPartner.vue", () => {
  const wrapper = shallowMount(FindPartnerPage, {
    global: {
      plugins: [Quasar],
    },
  });
  it("should find findPartner data-testid", () => {
    const findPartner = wrapper.find('[data-testid="findPartner"]');
    expect(findPartner.exists()).toBe(true);
  });

  it("should find findPartnerContent data-testid", () => {
    const findPartnerContent = wrapper.find(
      '[data-testid="findPartnerContent"]'
    );
    expect(findPartnerContent.exists()).toBe(true);
    expect(findPartnerContent.text()).toContain("Find");
    expect(findPartnerContent).toMatchSnapshot();
  });
});
