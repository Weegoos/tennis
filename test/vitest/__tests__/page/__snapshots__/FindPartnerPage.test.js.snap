// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`tests for FindPartner.vue > should find findPartnerContent data-testid 1`] = `
DOMWrapper {
  "isDisabled": [Function],
  "wrapperElement": <p
    class="text-bold text-h4"
    data-testid="findPartnerContent"
    data-v-27de1c0a=""
  >
     Find a partner 
  </p>,
}
`;
